be/src/service/staros_worker.cpp

```diff
@@ -159,79 +182,114 @@ absl::StatusOr<std::shared_ptr<fslib::FileSystem>> StarOSWorker::build_filesyste
+    auto cache_info = info.cache_info;
+    bool cache_enabled = cache_info.enable_cache() && !cache_dir.empty();
+    if (cache_enabled) {
+        return "cachefs://";
+    }
     std::string scheme = "file://";
     switch (info.path_info.fs_info().fs_type()) {
     case staros::FileStoreType::S3:
         scheme = "s3://";
-        {
-            auto& s3_info = info.path_info.fs_info().s3_fs_info();
-            if (!info.path_info.full_path().empty()) {
-                localconf[fslib::kSysRoot] = info.path_info.full_path();
-            }
-            if (!s3_info.bucket().empty()) {
-                localconf[fslib::kS3Bucket] = s3_info.bucket();
-            }
-            if (!s3_info.region().empty()) {
-                localconf[fslib::kS3Region] = s3_info.region();
-            }
-            if (!s3_info.endpoint().empty()) {
-                localconf[fslib::kS3OverrideEndpoint] = s3_info.endpoint();
-            }
-            if (s3_info.has_credential()) {
-                auto credential = s3_info.credential();
-                if (credential.has_default_credential()) {
-                    localconf[fslib::kS3CredentialType] = "default";
-                } else if (credential.has_simple_credential()) {
-                    localconf[fslib::kS3CredentialType] = "simple";
-                    auto simple_credential = credential.simple_credential();
-                    localconf[fslib::kS3CredentialSimpleAccessKeyId] = simple_credential.access_key();
-                    localconf[fslib::kS3CredentialSimpleAccessKeySecret] = simple_credential.access_key_secret();
-                } else if (credential.has_profile_credential()) {
-                    localconf[fslib::kS3CredentialType] = "instance_profile";
-                } else if (credential.has_assume_role_credential()) {
-                    localconf[fslib::kS3CredentialType] = "assume_role";
-                    auto role_credential = credential.assume_role_credential();
-                    localconf[fslib::kS3CredentialAssumeRoleArn] = role_credential.iam_role_arn();
-                    localconf[fslib::kS3CredentialAssumeRoleExternalId] = role_credential.external_id();
-                } else {
-                    localconf[fslib::kS3CredentialType] = "default";
-                }
-            }
-        }
         break;
     case staros::FileStoreType::HDFS:
         scheme = "hdfs://";
-        localconf[fslib::kSysRoot] = info.path_info.full_path();
         break;
     case staros::FileStoreType::AZBLOB:
         scheme = "azblob://";
-        {
+        break;
+    default:
+        return absl::InvalidArgumentError("Unknown shard storage scheme!");
+    }
+    return scheme;
+}
+
+absl::Status StarOSWorker::build_conf_from_shard_info(const ShardInfo& info, Configuration& localconf) {
+    // FIXME: currently the cache root dir is set from be.conf, could be changed in future
+    std::string cache_dir = config::starlet_cache_dir;
```

Comment:
looks like we can extract these lines into a small static function `need_enable_cache(CacheInfo)`